staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -267,6 +319,44 @@ type PrefixedClaimOrExpression struct {
 	// prefix is prepended to claim's value to prevent clashes with existing names.
 	// +required
 	Prefix *string `json:"prefix"`
+
+	// expression represents the expression which will be evaluated by CEL.
+	// Must produce a string. CEL expressions have access to the contents of the
+	// token claims for claimValidationRules and claimMappings, userInfo for userInfoValidationRules.
+	// Documentation on CEL: https://kubernetes.io/docs/reference/using-api/cel/
+	// Either claim or expression must be set.
+	// +optional
+	Expression *string `json:"expression"`
+}
+
+type ClaimOrExpression struct {
+	// claim is the JWT claim to use.
+	// Either claim or expression must be set.
+	// +optional
+	Claim string `json:"claim"`
+
+	// expression represents the expression which will be evaluated by CEL.
+	// Must produce a string. CEL expressions have access to the contents of the
+	// token claims for claimValidationRules and claimMappings.
+	// TODO(aramase): think about what happens if the claim is absent or the wrong type
+	Expression string `json:"expression"`
+}
+
+type ExtraMapping struct {
+	// key is a string to use as the extra attribute key.
+	Key string `json:"key"`
+	// valueExpression is a CEL expression to extract extra attribute value.
+	// valueExpression must produce a string or string array value.
+	// "", [], and null values are treated as the extra mapping not being present.
+	// Empty string values contained within a string array are filtered out.
```

Comment:
Instead of filtering, should we require that the CEL author do whatever they need to send back a valid array? I.e. `mylist.filter(x, x != "")`

The filtering may mask errors and make it harder to debug CEL expressions than if we just produce an error explaining what went wrong